themes/cypress/source/js/toc.js

```diff
@@ -9,6 +9,20 @@
 
   if (!toc) return;
 
+  // override MenuSpy's .tick implementation because it incorrectly
+  // calculates the current scroll offset and doesn't handle the
+  // case where the user is right at the top of the document.
+  MenuSpy.prototype.tick = function tick () {
+    var fromTop = this.currScrollTop + headerHeight;
+    var inViewElms = this.scrollItems.filter(function (item) {
+      return item.offset < fromTop;
+    });
+    var elementInView = (inViewElms.length > 0)
+      ? inViewElms.pop()
+      : this.scrollItems[0];
+    this.activateItem(elementInView)
+  };
```

Comment:
Honestly, this feels like a band-aid solution 😳

The better (?) approach would be to address this at the source in the MenuSpy library. I did start down that route but my enthusiasm was swiftly extinguished when I discovered that [there are no tests of any significance](https://github.com/lcdsantos/menuspy/blob/master/test/basic.spec.js) in the MenuSpy source code.

> I'm not code shaming there: libraries sans tests are fine... I _personally_ can't be bothered investing my time into them though.